app.rb

```diff
@@ -13,7 +13,17 @@
   erb :start
 end
 
+get '/trip'
+@num_stops =
+erb :trip
+end
+
 get '/end' do
   @stations = metro[params[:line].to_sym]
   erb :end
 end
+
+get '/trip'
+@num_stops =
```

Comment:
We were looking for you to do a few things here. First, we need to get parameters from the url.

With no named variables in our path like `'/artists/:id'` we can instead get them from query parameters like `'/?artist_id=1&song_id=5'`

In this assignment, we expect to get a url like this:
`'http://localhost:4567/trip?line=green&start=U+Street&end=Waterfront'`

Which gives us a `params` hash:
```ruby
params = {
    line: "green",
    start: "U Street",
    end: "Waterfront"
} 
```

Inside of our `trip` route we access the `params` hash and the `metro` hash (provided by the `metro.rb` file).

In particular we'll take the line color `params[:line]`, and convert it to a symbol `params[:line].to_sym`. And that will be the key we use to get the list of stations on that line:
```ruby
stations = metro[params[:line].to_sym]
# the stations are assumed to be the same order as you would travel
```

And find the index of the starting and ending stations:
```ruby
start_index = stations.index(params[:start])
end_index = stations.index(params[:end])
```

These indexes are numbers between 0 and the total stations, so to get the trip length we just need to subtract the end from the start, and display the result:
```ruby
@num_stops = end_index - start_index
# in case we're starting at station number higher than our end, make sure to keep the trip length positive:
@num_stops*= -1 if(@num_stops < 0) 
```

This is the basic idea